app/controllers/appropriate_bodies/claim_an_ect/register_ect_controller.rb

```diff
@@ -5,18 +5,17 @@ def edit
         @pending_induction_submission = find_pending_induction_submission
       end
 
+      # OPTIMIZE: Pre-model validations are required to guard against
+      # ActiveRecord::MultiparameterAssignmentErrors
+      # when dates are non-existent or incorrect dates when one day from the month's end
+      # for example 31st Sept becomes 1st Oct
+      # or when negative numbers are passed in by mistake
       def update
-        register_ect = AppropriateBodies::ClaimAnECT::RegisterECT
-          .new(
-            appropriate_body: @appropriate_body,
-            pending_induction_submission: find_pending_induction_submission,
-            author: current_user
-          )
-
-        if register_ect.register(update_params)
+        if hash_date.valid? && register_ect.register(update_params)
```

Comment:
I think this validation logic should be in the `started_on` field in `PendingInductionSubmission` model instead of here in the controller.

Reply:
It should but it doesn't catch the error that way. It is intentionally pre-validating before the params are submitted.